### PR TITLE
Add eight more test problems/families

### PR DESCRIPTION
**Pull request by anriseth**
This adds a selection of quadratic problems and a subset of the problems from 
Moré JJ, Garbow BS, Hillstrom KE: Testing unconstrained optimization software. ACM T Math Software. 1981

As part of this PR, I have also introduced a parameter functionality for the problems.

### CODE REVIEW

**Comment by pkofod** [comment]
Great! Exactly what I meant wrt parameters

**Comment by anriseth** [comment]
Good:) I've used input dimension 100 on all the example problems here. I
don't know whether that's to high for Travis of we hook this up with Optim
CI?

**Comment by pkofod** [comment]
I think our travis run times are quite low right now (compared to the limit) so probably it will be fine. Anyway, it shouldn't be hard to change to a lower number without creating package administration chaos :)

**Comment by pkofod** [comment]
just tell me when this is ready

File: src/optim_tests/multivariate/quad_transforms.jl [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        retval = F[:Q]'*F[:Q]
+    end
+    retval
+end
+
+srand(0) # TODO: is it dodgy to set the random seed in a package?
+examples["Paraboloid Random matrix"] = _paraboloidproblem(100;
+                                                          name = "Paraboloid Random Matrix (100)",
+                                                          mat = _randommatrix(100))
+srand()  # TODO: is it fine if we reset the seed after?
```
Comment (anriseth) [inline src/optim_tests/multivariate/quad_transforms.jl]: Is this (and L137) bad practice? Are there situations where calling `srand` via `using` is problematic?

**Review by anriseth — COMMENTED** [review]
Good to go, modulo the `srand` question in the code.

File: src/optim_tests/multivariate/quad_transforms.jl (hunk 1 above)
Comment (pkofod) [reply src/optim_tests/multivariate/quad_transforms.jl]: Does it affect code outside of the module's scope?

**Comment by anriseth** [comment]
Only if they rely on random numbers in a weird, deterministic, way

File: src/optim_tests/multivariate/quad_transforms.jl (hunk 1 above)
Comment (pkofod) [reply src/optim_tests/multivariate/quad_transforms.jl]: Hmm, from trying this out, I'm not really sure how comfortable I am with this. I am a bit worried that even if it's only us who are using this, we might screw this up when using random numbers to benchmark or something similar. Can we not just store the matrix in explicit form? I know it's a bit... but...

**Comment by anriseth** [comment]
For benchmarks, I think people should set an explicit seed in any case.
I'll ask on Discourse or similar what guidelines are.

Adding it explicitly is quite simple as well, even if it looks I for
someone checking out the code in future. Or we could just leave this
particular problem out of the test set?

**Comment by pkofod** [comment]
Yes but it's quite normal to set your seed at the top level, even before loading modules, and then this will overwrite that seed

**Comment by anriseth** [comment]
Hmm, okay. That sounds like a bad habit, in my opinion. But if people do
this, then I'll change toys so we don't mess with the seed.

**Comment by pkofod** [comment]
> That sounds like a bad habit, in my opinion.

Well, yeah... Again, this is basically "our" module, so we can do whatever we want. Maybe you can capture the current random generator state and restore it?

**Comment by pkofod** [comment]
I've asked on slack, let's see what they say

**Comment by pkofod** [comment]
img_2295

**Comment by anriseth** [comment]
Thanks! The random number generator is now safe :p

**Comment by pkofod** [comment]
Merge it! Meeerge it!

**Pull request merged**